public/pages/CreateMonitor/components/MonitorExpressions/expressions/WhereExpression.js

```diff
@@ -0,0 +1,228 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { PureComponent, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'formik';
+import { EuiFlexGroup, EuiFlexItem, EuiPopover, EuiExpression, EuiText } from '@elastic/eui';
+import _ from 'lodash';
+import {
+  Expressions,
+  POPOVER_STYLE,
+  EXPRESSION_STYLE,
+  WHERE_BOOLEAN_FILTERS,
+} from './utils/constants';
+import {
+  getOperators,
+  displayText,
+  validateRange,
+  isNullOperator,
+  isRangeOperator,
+} from './utils/whereHelpers';
+import { isInvalid, required } from '../../../../../utils/validate';
+import {
+  FormikComboBox,
+  FormikSelect,
+  FormikFieldNumber,
+  FormikFieldText,
+} from '../../../../../components/FormControls';
+import { getIndexFields } from './utils/dataTypes';
+import { FORMIK_INITIAL_VALUES } from '../../../containers/CreateMonitor/utils/constants';
+import { DATA_TYPES } from '../../../../../utils/constants';
+
+const propTypes = {
+  formik: PropTypes.object.isRequired,
+  dataTypes: PropTypes.object.isRequired,
+  onMadeChanges: PropTypes.func.isRequired,
+  openedStates: PropTypes.object.isRequired,
+  openExpression: PropTypes.func.isRequired,
+};
+
+class WhereExpression extends PureComponent {
```

Comment:
Do we want this to be a PureComponent?
If we make a change to one of the fields in formik.values is a whole new "formik" object created or just the values updated?